jetty-http/src/main/java/org/eclipse/jetty/http/HttpGenerator.java

```diff
@@ -384,25 +384,28 @@ public Result generateResponse(MetaData.Response info, boolean head, ByteBuffer
                     // generate ResponseLine
                     generateResponseLine(info, header);
 
-                    // Handle 1xx and no content responses
-                    int status = info.getStatus();
-                    if (status >= 100 && status < 200)
+                    if (!info.isServletUpgrade())
```

Comment:
Can you explain this change a little bit more?

Reply:
I modified the impl according to our discussion on the matter.